### ISSUE TITLE
Very narrow python pyyaml requirement for parallelcluster 3.7.0

### ISSUE THREAD

**Issue by skoloCFD** [aws/aws-parallelcluster#5714]
pyyaml==6.0.1 is quite a narrow requirement, and breaks compatibility with many other AWS packages, like boto3 and the awscli. Is such a strict requirement for pyyaml necessary? 6.0.1 was just released in July - it's unlikely to be compatible with anything this early on. Previous versions of parallelcluster were always much more flexible, and less "bleeding edge." Would be great to stick to that!

**Comment by lukeseawalker** [comment]
Hello @skoloCFD,
it makes totally sense, thanks for calling this out.

**Comment by skoloCFD** [comment]
When might this actually go into effect? I still get the same problem installing parallelcluster 3.7.2

The conflict is caused by:
    The user requested PyYAML==5.3.1
    aws-parallelcluster 3.7.2 depends on PyYAML==6.0.1

**Comment by enrico-usai** [comment]
Hi @skoloCFD what about using a [Python virtual environment](https://docs.aws.amazon.com/parallelcluster/latest/ug/install-v3-virtual-environment.html) specific for pcluster?
This will save you from compatibility issues with other tools and python packages.

**Comment by skoloCFD** [comment]
That would be significantly more maintenance/effort than just having a requirements.txt with one set of AWS packages. This is especially true for deployments to managed AWS services where python packages are managed by AWS via requirements.txt.

pyyaml 6.0.1 is only a few months old and is incompatible with many other AWS python modules, so I would conversely ask, is 6.0.1 absolutely necessary for pc3.7?

**Comment by enrico-usai** [comment]
Thanks for your answer.

In general, using a [virtual environment](https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#create-and-use-virtual-environments) is a good practice.

Did you try to uninstall 6.0.1 and install 5.3.1?
``` 
$ pip uninstall PyYAML
$ pip install PyYAML==5.3.1
```

Anyway we fixed the [requirements](https://github.com/aws/aws-parallelcluster/blob/develop/cli/requirements.txt#L25) by removing the 6.0.1 for the upcoming ParallelCluster 3.8.0 release (3.8.0b1 is already available).
New requirement for ParallelCluster will be: `PyYAML>=5.3.1,!=5.4` so PyYAML 5.3.1 will be considered valid.

**Comment by skoloCFD** [comment]
Yes, I had PyYAML-5.3.1 installed (with parallelcluster3.6), but as soon as parallelcluster3.7 is installed, 5.3.1 is removed and PyYAML-6.0.1 gets installed. I can see this comes from a requirement of parallecluster:
`Collecting PyYAML==6.0.1 (from aws-parallelcluster==3.7.2)`

For the meantime I will stick to parallelcluster 3.6, and am happy to see that PyYAML-6.0.1 will not be necessary for parallelcluster3.8. When 3.8 is released I will give this another try. Thanks!

**Comment by enrico-usai** [comment]
The workaround I was suggesting was to uninstall 6.0.1 and install 5.3.1 after the installation of ParallelCluster:
```
$ pip install aws-parallelcluster
...Collecting PyYAML==6.0.1 (from aws-parallelcluster==3.7.2)
$ pip uninstall PyYAML
$ pip install PyYAML==5.3.1
```

`pcluster` command should continue to work because the code hasn't a strict dependency on that version of PyYAML.

**Comment by skoloCFD** [comment]
Ah ok, I see what you mean. This will work on a local system, but on AWS managed services where python is built automatically, it will fall over due the the warnings thrown by pip. However, I can confirm the issue goes away with 3.8.0b1
`Successfully installed PyYAML-5.3.1 aws-parallelcluster-3.8.0b1`

So it looks like my issue will be solved by waiting for 3.8, and in the meantime I can use a version below 3.7. Appreciate the replies!

**Comment by enrico-usai** [comment]
Thank you for confirming that 3.8.0b1 will resolve your issue and for describing your use case!

**Comment by enrico-usai** [comment]
Fix released as part of [3.8.0](https://github.com/aws/aws-parallelcluster/releases/tag/v3.8.0)
